src/rust/aggregator/src/data/leaderboards.rs

```diff
@@ -0,0 +1,261 @@
+                        SELECT array_agg(integrator) AS integrators FROM (
+                            SELECT DISTINCT "user", integrator FROM place_limit_order_events
+                            WHERE NOT EXISTS (
+                                SELECT * FROM competition_indexed_events
+                                WHERE place_limit_order_events.txn_version = competition_indexed_events.txn_version
+                                AND place_limit_order_events.event_idx = competition_indexed_events.event_idx
+                                AND competition_indexed_events.competition_id = $2
+                            )
+                            AND time > $3 AND time < $4
+                            UNION
+                            SELECT DISTINCT "user", integrator FROM place_market_order_events
+                            WHERE NOT EXISTS (
+                                SELECT * FROM competition_indexed_events
+                                WHERE place_market_order_events.txn_version = competition_indexed_events.txn_version
+                                AND place_market_order_events.event_idx = competition_indexed_events.event_idx
+                                AND competition_indexed_events.competition_id = $2
+                            )
+                            AND time > $3 AND time < $4
+                            UNION
+                            SELECT DISTINCT signing_account as "user", integrator FROM place_swap_order_events
+                            WHERE NOT EXISTS (
+                                SELECT * FROM competition_indexed_events
+                                WHERE place_swap_order_events.txn_version = competition_indexed_events.txn_version
+                                AND place_swap_order_events.event_idx = competition_indexed_events.event_idx
+                                AND competition_indexed_events.competition_id = $2
+                            )
+                            AND time > $3 AND time < $4
+                        ) AS a
+                        WHERE integrator IN (SELECT unnest(frontends_required) FROM competition_metadata)
+                        AND "user" = $1
+                    ),
+                    trades = (
+                        WITH homogenous_fills AS (
+                            SELECT taker_address AS "user", taker_order_id AS order_id
+                            FROM fill_events
+                            WHERE time > $3 AND time < $4
+                            UNION
+                            SELECT maker_address AS "user", maker_order_id AS order_id
+                            FROM fill_events
+                            WHERE time > $3 AND time < $4
+                        )
+                        SELECT COUNT(order_id) AS orders
+                        FROM homogenous_fills
+                        WHERE "user" = $1
+                    )
+                WHERE "user" = $1
+            "#,
```

Comment:
Let's abstract this out

If something doesn't work it will be a headache to find the error